clientd3d/server.c

```diff
@@ -351,33 +353,7 @@ void ExtractObject(char **ptr, object_node *item)
 
 void ExtractObjectNoLight(char **ptr, object_node *item)
 {  
-   Extract(ptr, &item->id, SIZE_ID);
-   if (IsNumberObj(item->id))
-      Extract(ptr, &item->amount, SIZE_AMOUNT);
-   else
-      item->amount = 0;
-   Extract(ptr, &item->icon_res, SIZE_ID);
-   Extract(ptr, &item->name_res, SIZE_ID);
-   Extract(ptr, &item->flags, 4); // includes drawfx_mask bits
-
-   ExtractPaletteTranslation(ptr,&item->translation,&item->effect);
-   item->normal_translation = item->translation;
-   item->secondtranslation = XLAT_FILTERWHITE90;
-
-   ExtractAnimation(ptr, &item->normal_animate);
-   item->animate = &item->normal_animate;
-
-   item->normal_overlays = ExtractOverlays(ptr);
-   item->overlays = &item->normal_overlays;
-
-   if (OF_BOUNCING == (OF_BOUNCING & item->flags))
-   {
-      item->bounceTime = (WORD)(rand() % 1000);
-   }
-   if (OF_PHASING == (OF_PHASING & item->flags))
-   {
-      item->phaseTime = (WORD)(rand() % 1000);
-   }
+   ExtractObject(ptr, item, False); // Disable light extraction
```

Comment:
bool has the values true/false, not True/False

Reply:
This one really caught me, because I checked beforehand.  The capitalization seemed off.

However, there are so many instances in this file with capitalization that I wasn't sure what to use:
- 78 instances of `return True;`
- 84 instances of `return False;`

It looks like there is an enum in `client.h` for these capitalized booleans
- `enum {False = 0, True = 1};`

I don't mind changing to the lowercase values of `true` and `false`.  Should I do that here?  Or keep the capitalized values so they're the consistent within this file?

Reply:
The short answer is there are two different types.  There is "Bool" that has the values True/False, and "bool" that has the values true/false.  Don't mix them.

The longer answer is that when Meridian was written, there wasn't a good C++ compiler, so rather we were unable to use C++'s builtin "bool" type, and instead we defined our own "Bool" type (that enum you found).  Today we would not do this, but instead we'd simply use "bool" everywhere.  New code should use bool.

Windows is even older, and has its own BOOL type that has the values TRUE and FALSE.  This is to be avoided except when dealing with Windows calls that require it.

Reply:
Thank you for the explanation!  I adjusted the `bool` parameter for `ExtractObject` to use `true` and `false` (lowercase).

This was changed in 2 places:
-  the `server.h` file for the default `bool` value of `true`
- `server.c`'s wrapper function `ExtractObjectNoLight` with `bool` parameter `false`